server/src/main/java/io/unitycatalog/server/service/ModelService.java

```diff
@@ -3,102 +3,260 @@
+          #authorize(#principal, #metastore, OWNER) ||
+          #authorize(#principal, #registered_model, OWNER) ||
+          (#authorize(#principal, #registered_model, EXECUTE) && #authorize(#principal, #schema, USE_SCHEMA) && #authorize(#principal, #catalog, USE_CATALOG))
+          """)
+  @AuthorizeKey(METASTORE)
   public HttpResponse getModelVersion(
-      @Param("full_name") String fullName, @Param("version") Long version) {
+      @Param("full_name") @AuthorizeKey(REGISTERED_MODEL) String fullName, @Param("version") Long version) {
     assert fullName != null && version != null;
     ModelVersionInfo modelVersionInfo = MODEL_REPOSITORY.getModelVersion(fullName, version);
     return HttpResponse.ofJson(modelVersionInfo);
   }
 
   @Patch("/{full_name}/versions/{version}")
-  public HttpResponse updateModelVersion(UpdateModelVersion updateModelVersion) {
+  @AuthorizeExpression("""
+          #authorize(#principal, #metastore, OWNER) ||
+          #authorize(#principal, #catalog, OWNER) ||
+          (#authorize(#principal, #catalog, USE_CATALOG) && #authorize(#principal, #schema, OWNER)) ||
+          (#authorize(#principal, #registered_model, OWNER) && #authorize(#principal, #schema, USE_SCHEMA) && #authorize(#principal, #catalog, USE_CATALOG))
+          """)
+  @AuthorizeKey(METASTORE)
+  public HttpResponse updateModelVersion(@Param("full_name") @AuthorizeKey(REGISTERED_MODEL) String fullName,
+                                         UpdateModelVersion updateModelVersion) {
     assert updateModelVersion != null;
     ModelVersionInfo updateModelVersionResponse =
         MODEL_REPOSITORY.updateModelVersion(updateModelVersion);
     return HttpResponse.ofJson(updateModelVersionResponse);
   }
 
   @Delete("/{full_name}/versions/{version}")
+  @AuthorizeExpression("""
+          #authorize(#principal, #metastore, OWNER) ||
+          (#authorize(#principal, #registered_model, OWNER) && #authorize(#principal, #schema, USE_SCHEMA) && #authorize(#principal, #catalog, USE_CATALOG))
+          """)
+  @AuthorizeKey(METASTORE)
   public HttpResponse deleteModelVersion(
-      @Param("full_name") String fullName, @Param("version") Long version) {
+      @Param("full_name") @AuthorizeKey(REGISTERED_MODEL) String fullName, @Param("version") Long version) {
     MODEL_REPOSITORY.deleteModelVersion(fullName, version);
     return HttpResponse.of(HttpStatus.OK);
   }
 
   @Patch("/{full_name}/versions/{version}/finalize")
-  public HttpResponse finalizeModelVersion(FinalizeModelVersion finalizeModelVersion) {
+  @AuthorizeExpression("""
+          #authorize(#principal, #metastore, OWNER) ||
+          #authorize(#principal, #catalog, OWNER) ||
+          (#authorize(#principal, #catalog, USE_CATALOG) && #authorize(#principal, #schema, OWNER)) ||
+          (#authorize(#principal, #registered_model, OWNER) && #authorize(#principal, #schema, USE_SCHEMA) && #authorize(#principal, #catalog, USE_CATALOG))
+          """)
```

Comment:
I think I misspoke on this.  Only the RegisteredModel owner should be able to finalize a model version since they are the only ones who can create a model version.

Reply:
Done.